skops/io/_general.py

```diff
@@ -11,12 +11,16 @@ def dict_get_state(obj, dst):
         "__class__": obj.__class__.__name__,
         "__module__": get_module(type(obj)),
     }
+
+    # Since json converts dict keys to strings, we store the dict 4 arrays:
+    # keys, key-types, values, value-types.
     content = {}
-    for key, value in obj.items():
-        if np.isscalar(key) and hasattr(key, "item"):
-            # convert numpy value to python object
-            key = key.item()
-        content[key] = _get_state(value, dst)
+    keys = np.array(list(obj.keys()))
```

Comment:
Hmm, I don't have a good feeling about using numpy within the general module. No concrete issue but rather it smells wrong. Would it be possible to solve the problem when making the assumption that the types of all keys are the same, and then storing the key type as an extra value on the dict? And if there are different types, raise an error for the time being?

Reply:
Fair point, I rolled it back, now only save key types, and it solves a few issues actually.